[BUGFIX release] Ensure query param only link-to's work in error states.

2hu12 opened the pull request:
Try to fix https://github.com/emberjs/ember.js/issues/17963, I'm not sure this is the right way and it's weird, but the old "qualifiedRouteName" property works in this way and "_route" is used similarly to "qualifiedRouteName".

2hu12 committed:
[BUGFIX release] Ensure query param only link-to's work in error states.

Try to fix https://github.com/emberjs/ember.js/issues/17963

rwjblue reviewed (commented):
The fix seems reasonable (thank you!), but I think we need to add a previously failing test to ensure we don't regress again...

2hu12 commented:
I try to add one test in "packages/@ember/-internals/glimmer/tests/integration/components/link-to/query-params-curly-test.js", but with no success. The test need an `{{#link-to}}` with no route parameter and the bug could be reproduced when transition to an error route which might have a rejected model, but when it encounters an error like `model() { RSVP.reject(new Error('some error')) }` in test, the whole test just terminated.

Do you have any idea on how to write this previously failing test? Thank you!

villander commented:
@rwjblue @dgeb  any idea about this? We are with ember-engines CI broken.

I'll be happy if you can add any tip to finish this, thanks

villander commented:
we are trying fix this on engines side as well - https://github.com/ember-engines/ember-engines/pull/647

villander commented:
@2hu12 unfortunately I don't was able to write tests for this  =/

dgeb committed:
[BUGFIX release] Add regression test for QP only link-to inside error route.

Reproduce failure in #17963: `<LinkTo>` without `@route` param but with
`@query` breaks when navigating to error route.

Co-authored-by: Robert Jackson <me@rwjblue.com>

rwjblue commented:
I just rebased and pushed a previously failing test (using the one @dgeb started over in #18061), once CI is green we should be good to go.

villander commented:
thanks @rwjblue <3
